num/dual/dual_test.go

```diff
@@ -387,6 +387,74 @@ func TestPowReal(t *testing.T) {
+	{d: Number{Real: math.NaN(), Emag: 3}, want: Number{Real: math.NaN(), Emag: 3}},
+	{d: Number{Real: math.NaN(), Emag: 1}, want: Number{Real: math.NaN(), Emag: 1}},
+	{d: Number{Real: math.NaN(), Emag: math.Inf(1)}, want: Number{Real: math.NaN(), Emag: math.Inf(1)}},
+	{d: Number{Real: math.NaN(), Emag: math.Inf(-1)}, want: Number{Real: math.NaN(), Emag: math.Inf(-1)}},
+
+	// Abs(x+xϵ) = 0+0ϵ for any x = ±0
+	{d: Number{Real: negZero, Emag: negZero}, want: Number{Real: 0, Emag: negZero}},
+	{d: Number{Real: negZero, Emag: 0}, want: Number{Real: 0, Emag: 0}},
+	{d: Number{Real: 0, Emag: negZero}, want: Number{Real: 0, Emag: 0}},
+	{d: Number{Real: 0, Emag: 0}, want: Number{Real: 0, Emag: 0}},
+
+	// Abs(x+NaNϵ) = |x|+NaNϵ for any x
+	{d: Number{Real: -1, Emag: math.NaN()}, want: Number{Real: 1, Emag: math.NaN()}},
+	{d: Number{Real: 1, Emag: math.NaN()}, want: Number{Real: 1, Emag: math.NaN()}},
+	{d: Number{Real: 10, Emag: math.NaN()}, want: Number{Real: 10, Emag: math.NaN()}},
+
+	// Abs(±Infinity+NaNϵ) = (Infinity+NaNϵ)
+	{d: Number{Real: math.Inf(-1), Emag: 0}, want: Number{Real: math.Inf(1), Emag: 0}},
+	{d: Number{Real: math.Inf(1), Emag: 0}, want: Number{Real: math.Inf(1), Emag: 0}},
+
+	// Abs(1 + yϵ) = (1 + yϵ) for any y
+	{d: Number{Real: 1, Emag: -3}, want: Number{Real: 1, Emag: -3}},
+	{d: Number{Real: 1, Emag: -2}, want: Number{Real: 1, Emag: -2}},
+	{d: Number{Real: 1, Emag: -1}, want: Number{Real: 1, Emag: -1}},
+	{d: Number{Real: 1, Emag: 0}, want: Number{Real: 1, Emag: 0}},
+	{d: Number{Real: 1, Emag: 1}, want: Number{Real: 1, Emag: 1}},
+	{d: Number{Real: 1, Emag: 2}, want: Number{Real: 1, Emag: 2}},
+	{d: Number{Real: 1, Emag: 3}, want: Number{Real: 1, Emag: 3}},
+	{d: Number{Real: 1, Emag: math.NaN()}, want: Number{Real: 1, Emag: math.NaN()}},
+	{d: Number{Real: 1, Emag: math.Inf(-1)}, want: Number{Real: 1, Emag: math.Inf(-1)}},
+	{d: Number{Real: 1, Emag: math.Inf(1)}, want: Number{Real: 1, Emag: math.Inf(1)}},
+
+	// AbsReal(Infinity+xϵ) = 0+NaNϵ for |x| < 1
+	{d: Number{Real: math.Inf(1), Emag: 0}, want: Number{Real: math.Inf(1), Emag: 0}},
+	{d: Number{Real: math.Inf(1), Emag: 0.1}, want: Number{Real: math.Inf(1), Emag: 0.1}},
+	{d: Number{Real: math.Inf(1), Emag: 0.2}, want: Number{Real: math.Inf(1), Emag: 0.2}},
+	{d: Number{Real: math.Inf(1), Emag: 0.5}, want: Number{Real: math.Inf(1), Emag: 0.5}},
+
+	// AbsReal(±x+yϵ) = |x|+NaNϵ for 0 < |x| < 1
+	{d: Number{Real: 1.175495e-38, Emag: math.NaN()}, want: Number{Real:1.175495e-38, Emag:math.NaN()}},
+	{d: Number{Real: -1.175495e-38, Emag: math.NaN()}, want: Number{Real:1.175495e-38, Emag:math.NaN()}},
+	{d: Number{Real: -0.2, Emag: math.NaN()}, want: Number{Real: 0.2, Emag: math.NaN()}},
+	{d: Number{Real: -0.1, Emag: math.NaN()}, want: Number{Real: 0.1, Emag: math.NaN()}},
+	{d: Number{Real: 0.1, Emag: math.NaN()}, want: Number{Real: 0.1, Emag: math.NaN()}},
+	{d: Number{Real: 0.2, Emag: math.NaN()}, want: Number{Real: 0.2, Emag: math.NaN()}},
+
```

Comment:
Please delete this blank line.

Reply:
> Please squash out the unnecessary commits here and remove the vim swap files.
> 
> It would be helpful if you provided the sources of the special case claims you are making in the tests.

Sorry about that. Thanks for your time. My intention is to add test coverage for functions discussed in #1093.

I just wanted to include edge and corner cases.

Reply:
We still need provenance. The current code matches exactly the reference implementation for dual and is trivially correct for the dual complex according to the reference paper.